src/components/DataTable/DataTable.tsx

```diff
@@ -230,99 +158,48 @@ class DataTable extends React.PureComponent<CombinedProps, DataTableState> {
                 {totalsMarkup}
               </thead>
               <tbody>{bodyMarkup}</tbody>
-              {footerMarkup}
             </table>
           </div>
+          {footerMarkup}
         </div>
       </div>
     );
   }
 
-  private tallestCellHeights = () => {
-    const {footerContent, truncate} = this.props;
-    const {
-      table: {current: table},
-    } = this;
-    let {heights} = this.state;
-    if (table) {
-      const rows = Array.from(table.getElementsByTagName('tr'));
-
-      if (!truncate) {
-        return (heights = rows.map((row) => {
-          const fixedCell = (row.childNodes as NodeListOf<HTMLElement>)[0];
-          return Math.max(row.clientHeight, fixedCell.clientHeight);
-        }));
-      }
-
-      if (footerContent) {
-        const footerCellHeight = (rows[rows.length - 1]
-          .childNodes as NodeListOf<HTMLElement>)[0].clientHeight;
-        heights = [footerCellHeight];
-      }
-    }
-
-    return heights;
-  };
-
-  private resetScrollPosition = () => {
-    const {
-      scrollContainer: {current: scrollContainer},
-    } = this;
-    if (scrollContainer) {
-      const {
-        preservedScrollPosition: {left, top},
-      } = this.state;
-      if (left) {
-        scrollContainer.scrollLeft = left;
-      }
-      if (top) {
-        window.scrollTo(0, top);
-      }
-    }
-  };
-
-  private setHeightsAndScrollPosition = () => {
```

Comment:
The scroll position needed to be preserved when the row heights were being managed manually because the horizontal scroll position would be reset on rerender (most notably after clicking a sortable cell heading). This is no longer an issue.